tests/cupy_tests/core_tests/test_dlpack.py

```diff
@@ -74,6 +89,96 @@ def test_conversion(self, dtype):
         testing.assert_array_equal(
             orig_array.data.ptr, out_array.data.ptr)
 
+    def test_from_dlpack_and_conv_errors(self):
+        orig_array = _gen_array("int8")
+
+        with pytest.raises(NotImplementedError):
+            cupy.from_dlpack(orig_array, device=orig_array.device)
+
+        with pytest.raises(BufferError):
+            # `__dlpack__` only allows `copy=True` for host copies.
```

Comment:
Took me a while to think about this. Let's expand the comment a bit
```suggestion
            # Currently CuPy's `__dlpack__` only allows `copy=True`
            # for host copies.
```
(to emphasize it's not because the protocol disallows it, but because we don't want to overcomplicate it).